### ISSUE TITLE
Add baseline evaluation of pretrained model before fine-tuning

### ISSUE THREAD

**Issue by susannecoates** [susannecoates/meta-es-for-hyperparam-opt#1]
## Problem
The current implementation optimizes fine-tuning hyperparameters but doesn't evaluate the pretrained model's performance before fine-tuning. This makes it impossible to demonstrate how much the optimized hyperparameters improve over the zero-shot baseline.

## Proposed Solution
Add evaluation of the pretrained `google/mobilebert-uncased` model on each task's validation set before starting the Meta-ES optimization loop.

## Tasks
- [ ] Add baseline evaluation after dataset loading (line ~127)
- [ ] Implement baseline evaluation code (lines ~163-181)
- [ ] Update results summary to show baseline comparison (lines ~330-337)
- [ ] Save baseline metrics to JSON output (lines ~350-355)

## Implementation Details
See TODOs added in `experiment-1/experiment-1.py`.

**Code structure:**
```python
# After tokenization, before hyperparameter bounds
baseline_model = AutoModelForSequenceClassification.from_pretrained(
    model_name, num_labels=2
).to("cpu")
baseline_trainer = CPUTrainer(
    model=baseline_model,
    eval_dataset=tokenized_datasets["validation"].select(range(min(500, len(tokenized_datasets["validation"])))),
    compute_metrics=compute_metrics
)
baseline_metrics = baseline_trainer.evaluate()
baseline_accuracy = baseline_metrics["eval_accuracy"]
print(f"\n{'='*70}")
print(f"BASELINE (Pre-Fine-Tuning) Accuracy: {baseline_accuracy:.4f} ({baseline_accuracy*100:.2f}%)")
print(f"{'='*70}\n")
```

## Expected Outcome
- Console output shows baseline accuracy
- Final summary includes improvement over baseline
- Results JSON includes baseline metrics for comparison
- Paper can report: "The optimized configuration improved accuracy by X percentage points over the pretrained baseline"

## Related Files
- `experiment-1/experiment-1.py`
- Paper: `paper/meta-es-for-hyper-opt.tex` (Future Directions section already mentions this)
